integrationtests/tests/fixtures/mod.rs

```diff
@@ -1020,60 +1039,52 @@ impl FederationTest {
         Ok(())
     }
 
+    /// Does any of the modules return consensus proposal that forces a new epoch
     #[allow(clippy::await_holding_refcell_ref)]
     pub async fn has_pending_epoch(&self) -> bool {
         for server in &self.servers {
-            if server
-                .borrow_mut()
-                .fedimint
-                .consensus
-                .get_consensus_proposal()
-                .await
-                .items
-                .is_empty()
+            let mut server = server.borrow_mut();
+            let fedimint_server = &mut server.fedimint;
+
+            // Pending transactions will trigger an epoch
+            if let Some(Some(_)) = Pin::new(&mut fedimint_server.tx_receiver)
+                .peek()
+                .now_or_never()
             {
                 return true;
             }
+
+            let consensus_proposal = fedimint_server.consensus.get_consensus_proposal().await;
+            if consensus_proposal.force_new_epoch {
+                return true;
+            }
         }
         false
     }
 
-    /// Returns true if the fed would produce an empty epoch proposal (no new information)
-    async fn empty_proposal(&self, server: &mut FedimintServer) -> bool {
-        // Hack to avoid saying the proposal is empty if there are tx in the channel
-        if let Ok(tx) = server.tx_receiver.get_mut().as_mut().try_recv() {
-            server.consensus.tx_sender.send(tx).await.expect("Can send");
-            return false;
-        }
-
-        let wallet = server
-            .consensus
-            .modules
-            .get_expect(self.wallet_id)
-            .as_any()
-            .downcast_ref::<Wallet>()
-            .unwrap();
-        let mut dbtx = block_on(server.consensus.db.begin_transaction());
-        let mut module_dbtx = dbtx.with_module_prefix(self.wallet_id);
-        let height = block_on(wallet.consensus_height(&mut module_dbtx)).unwrap_or(0);
-        let proposal = block_on(server.consensus.get_consensus_proposal());
-
-        for item in proposal.items {
-            match item {
-                // ignore items that get automatically generated
-                ConsensusItem::Module(mci) => match mci.as_any().downcast_ref() {
-                    Some(WalletConsensusItem::RoundConsensus(rci))
-                        if rci.block_height == height =>
-                    {
-                        continue
-                    }
-                    _ => return false,
-                },
-                ConsensusItem::EpochOutcomeSignatureShare(_) => continue,
-                _ => return false,
+    /// Get the consensus items proposed by all the peers
+    ///
+    /// Notably, unlike [`has_pending_epoch`] this does not take into account
+    /// pending transactions or ignore consensus items that do not trigger
+    /// epoch on its own.
```

Comment:
confusing language. 
So, this method returns all proposals including pending transactions(?) and ConsensusItems that do not trigger an epoch on their own?

Reply:
https://github.com/fedimint/fedimint/pull/1634